src/main/java/frc/robot/subsystems/Elevator.java

```diff
@@ -50,48 +47,42 @@
 import frc.robot.Robot;
 
 public class Elevator extends SubsystemBase {
-  private SparkMax elevatorMotor;
-  private SparkAbsoluteEncoder elevatorEncoder;
-  private SparkMaxSim simElevatorMotor;
+  private TalonFX elevatorMotor1;
+  private TalonFX elevatorMotor2;
   private final Time sparkPeriod;
   private static Mechanism2d mech = new Mechanism2d(2, 2);
   private static ComplexWidget widg;
+  private DigitalInput elvLimitSwitch;
   private MechanismLigament2d elevatorMech;
 
   public static final AbsoluteEncoderConfig encoderCfg =
       new AbsoluteEncoderConfig()
           .positionConversionFactor(Constants.ElevatorConstants.kElevatorConversion);
-  public static final SoftLimitConfig elvMaxSoftLimit =
-      new SoftLimitConfig()
-          .forwardSoftLimit(heightToRotations(Constants.ElevatorConstants.kTopLimit).in(Rotations));
-  public static final SoftLimitConfig elvMinSoftLimit =
-      new SoftLimitConfig()
-          .reverseSoftLimit(
-              heightToRotations(Constants.ElevatorConstants.kBottomLimit).in(Rotations));
-  public static final ClosedLoopConfig loopCfg =
-      new ClosedLoopConfig()
-          .feedbackSensor(FeedbackSensor.kPrimaryEncoder)
-          .pidf(
-              Constants.ElevatorConstants.P,
-              Constants.ElevatorConstants.I,
-              Constants.ElevatorConstants.D,
-              Constants.ElevatorConstants.FF);
+  public static final SoftwareLimitSwitchConfigs elvSoftLimit =
+      new SoftwareLimitSwitchConfigs()
+          .withForwardSoftLimitEnable(true)
+          .withForwardSoftLimitThreshold(heightToRotations(Constants.ElevatorConstants.kTopLimit))
+          .withReverseSoftLimitEnable(true)
```

Comment:
This soft limit needs to be disabled when the limitHit Command is run. Then it needs to get re-enabled afterward.